### ISSUE TITLE
Overly restrictive allowed charset list creating a large amount of false positive 920600 and 920480 rule matchings

### ISSUE THREAD

**Issue by sandumjacob** [coreruleset/coreruleset#3896]
### Description

The allowed charsets in regex-assembly/include/allowed-charsets.ra synced with tx.allowed_request_content_type_charset appears to be overly restrictive and I couldn't find any documentation or discussion on why only these 4 charsets are allowed. 
```
iso-8859-1
iso-8859-15
utf-8
windows-1252
```

For example, utf-16 is not allowed but is a relatively common and valid charset. 

I am wondering what the justification is for the decision to only include these 4 charsets. As shown, there are a lot more valid charsets than just these 4: https://www.iana.org/assignments/character-sets/character-sets.xhtml

### Accept Headers charset - triggering rule 920600 with valid charset utf-16
**utf-8 does not trigger rule**
![image](https://github.com/user-attachments/assets/42da575f-503c-45fb-92e0-01d43f6dc90d)
**utf-16 triggers rule**
![image](https://github.com/user-attachments/assets/be5779ce-b932-4b12-9274-d0cc831b6075)

### Content-Type charset - triggering rule 920480 with valid charset utf-16
**utf-8**
![image](https://github.com/user-attachments/assets/0d7776ed-777a-4115-8b1a-c8129e052bd3)
**utf-16 triggers rule**
![image](https://github.com/user-attachments/assets/2033a1d2-61e5-443a-841e-09b3a5274c06)

### Your Environment
Shown above in Coraza WAF Sandbox with CRS, also triggered in a prod environment using CRS 4.3.0 in coraza-proxy-wasm.
#
### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by fzipi** [comment]
This is a well-known problem. In summary: there is a static file that doesn't change even if you edit 900280 in `crs-setup.conf`. The problem is that the regex in 920600 is generated from  https://github.com/coreruleset/coreruleset/blob/44b82683028188f770541567db760e24810e84ca/regex-assembly/include/allowed-charsets.ra#L4, which is fixed by nature.

There is no easy fix for it now, unless we include UTF-16 as default. I'm going to bring this up in the next CRS monthly meeting.

**Comment by RedXanadu** [comment]
In theory, if we add UTF-16 as a 'default allowed' character encoding:
Can we guarantee that all engines fully implement and understand UTF-16 (requests and responses)?

A change like this feels like it could be opening the door to trivial evasions.

Maybe it is a decision best left to the user to explicitly **add** additional character encodings if they require something exotic.
The user is best placed to determine whether their WAF engine / server can correctly interpret such character encodings.

---

### Proposal:

1. Add a new, clear warning note, something like:
   *"Users can add additional character sets if something more exotic is required. Warning: If the WAF engine is unable to fully and correctly decode an encoding, this can lead to a full request body or response body bypass."*
2. Add a new explanation note about the regex compilation, something like:
   *"Newly added characters sets must also be added to regex-assembly/include/allowed-charsets.ra and the associated regular expressions must be re-compiled."*

**Comment by EsadCetiner** [comment]
resolved by https://github.com/coreruleset/coreruleset/pull/4003
